src/model/implementation/i18n.ts

```diff
@@ -106,6 +124,23 @@ export class ModelI18n implements ModelComponent {
             }),
         );
     }
+
+    /**
+     * Returns all available localization providers.
+     *
+     * The derived languages "I18N_GENERIC" and "I18N_LOCALE" are not included.
+     */
+    private getAllLocalizationProviders(): Record<string, LocalizationProvider> {
+        return Array.from(this.languageLocalizationProvidersByLanguage.keys())
+            .filter((lang) => lang !== I18N_GENERIC && lang !== I18N_LOCALE)
+            .reduce((providers, lang) => {
+                const provider = this.languageLocalizationProvidersByLanguage.get(lang);
+                if (provider) {
+                    providers[lang] = provider;
+                }
+                return providers;
+            }, {} as Record<string, LocalizationProvider>);
+    }
```

Comment:
unnecessary use of `reduce` - you're always returning the accumulator, so you're not actually reducing. I think a simple `for` loop would be more readable. Maybe even loop over `entries` instead of `keys` so you don't need another lookup? You probably also don't need the truthiness check on `provider` then.

alternatively, you could also use `.entries()`, filter them by key, and then use `Object.fromEntries()` to re-build the object

Reply:
Yes you are right, it can be written much easier.

I adapted it to the last suggestion, using `.entries()` and `Object.fromEntries()`